diff64.c

```diff
@@ -97,9 +122,15 @@ static void differentiate(const uint64_t left_number, const uint64_t right_numbe
         snprintf(&(lower_result[result_index]),  write_size, "%*u", ELEMENT_LENGTH, right_bit);
     }
 
+    max_number_length =  get_max_number_length(left_number, right_number);
```

Comment:
Add this line to the declaration of max_number_length and make it const (also there is two spaces before the function call :P)